### DISCUSSION TITLE
Stuck packaging a header-only library for vcpkg, can you help me?

### DISCUSSION THREAD

**Discussion by parkertomatoes** [microsoft/vcpkg#40865] [Q&A] [answered]
I wrote a header-only library awhile back that I'd like to package with vcpkg and eventually make available as a port.
https://github.com/parkertomatoes/lttb-cpp

But I can't get past an error I'm receiving when I try to call `vcpkg install ...` to get the hash for my library.

I've followed these steps:
  1. Followed this tutorial to create a CMake configuration for the library: https://dominikberner.ch/cmake-interface-lib/
  2. Following this guide here... https://learn.microsoft.com/en-us/vcpkg/get_started/get-started-packaging?pivots=shell-bash
  3. Created my overlay directories at `/path/to/custom-overlay` and `/path/to/custom-overlay/lttb`
  4. Created `vcpkg.json`, `usage`, `portfile.cmake`: [lttb-overlay.zip](https://github.com/user-attachments/files/16922624/lttb-overlay.zip)
  5. Ran `vcpkg install lttb --overlay-ports=/path/to/custom-overlay`

That's where the trouble starts. When I run this command from my overlay directory, I get this error: 
```
error: In manifest mode, `vcpkg install` does not support individual package arguments.
```

When I run the command from some other directory, I get this error: 
```
/path/to/custom-overlay/lttb: info: installing overlay port from here
-- Downloading https://github.com/parkertomatoes/lttb-cpp/archive/1.0.0.tar.gz -> parkertomatoes-lttb-cpp-1.0.0.tar.gz...
...
error: Missing parkertomatoes-lttb-cpp-1.0.0.tar.gz and downloads are blocked by x-block-origin.
error: https://github.com/parkertomatoes/lttb-cpp/archive/1.0.0.tar.gz: curl failed to download with exit code 22
```

Is there anything I can try to get back on track?

Thanks

**Comment by Osyotr** [comment]
> error: Missing parkertomatoes-lttb-cpp-1.0.0.tar.gz and downloads are blocked by x-block-origin.

Check your environment. There's probably `X_VCPKG_ASSET_SOURCES` with `x-block-origin` somewhere.

**Reply by parkertomatoes** [reply]
Thank you for replying! I checked my environment variables and did not find X_VCPKG_ASSET_SOURCES set to anything.

Attached is the debug output of the vcpkg command. These are the only two lines that mention x-block-origin, and they seem to contradict:
```
[DEBUG] External asset downloads are allowed (x-block-origin is disabled)...
...
error: Missing parkertomatoes-lttb-cpp-1.0.0.tar.gz and downloads are blocked by x-block-origin.
```
[output.txt](https://github.com/user-attachments/files/16939523/output.txt)

But stepping back a bit, do I need to have a zipped archive like the one it's trying to download to package my library for vcpkg? If so, how could I create that?

**Reply by Osyotr** [reply, marked as answer]
Hmm you may be using invalid URL. The tag in your repo is `v1.0` but you are trying to download `1.0.0`.

**Reply by parkertomatoes** [reply]
That was it! It wasn't clear to me before that a) there has to be a release, and b) the tag has to exactly match the "version" field of vcpkg.json.

Thank you for your help

**Reply by Osyotr** [reply]
> a) there has to be a release

Not really, just a tag would do.

> b) the tag has to exactly match the "version" field of vcpkg.json.

Not really, it's just that if a tag contains value of the version field you can use "${VERSION}" variable to reduce some copy-paste.

**Reply by dg0yt** [reply]
> That was it! It wasn't clear to me before that a) there has to be a release, and b) the tag has to exactly match the "version" field of vcpkg.json.
> 
> Thank you for your help

Even a commit SHA would do. Anything which *exists*.

**Reply by parkertomatoes** [reply]
Got it, thanks for making it clear everyone!
